contracts/governance/Governor.sol

```diff
@@ -311,19 +305,20 @@ abstract contract Governor is Context, ERC165, EIP712, Nonces, IGovernor, IERC72
         if (targets.length != values.length || targets.length != calldatas.length || targets.length == 0) {
             revert GovernorInvalidProposalLength(targets.length, calldatas.length, values.length);
         }
-        if (_proposals[proposalId].core.voteStart != 0) {
+        if (_proposals[proposalId].voteStart != 0) {
             revert GovernorUnexpectedProposalState(proposalId, state(proposalId), bytes32(0));
         }
 
         uint256 snapshot = clock() + votingDelay();
         uint256 duration = votingPeriod();
 
-        _proposals[proposalId].core = ProposalCore({
+        _proposals[proposalId] = ProposalCore({
             proposer: proposer,
             voteStart: SafeCast.toUint48(snapshot),
             voteDuration: SafeCast.toUint32(duration),
             executed: false,
-            canceled: false
+            canceled: false,
+            eta: 0
```

Comment:
It's probably because this is now writing 2 slots. We should try writing this as separate assignments for each member.

Reply:
Like 
```solidity
ProposalCore storage proposal = _proposals[proposalId];
proposal.proposer = proposer;
proposal.voteStart = SafeCast.toUint48(snapshot);
proposal.voteDuration = SafeCast.toUint32(duration);
```
?

I thought that was the one thing not to do.

Reply:
I did the commit, it works. 

I feel like I'm no longer understand the compiler. All the things we used to do for gas optimisation now turn out to be less efficient than the stuff we moved away from.

Reply:
> I guess the compiler just got better

Seem like `eta` is in separate slot in storage. So plain struct initialization writes “0” to this slot, since you need to initialize `eta` field. In your new approach compiler does not touch this slot at all